src/reporter/reporter.py

```diff
@@ -115,26 +115,25 @@ def notify():
                'content.', 400
 
     payload = request.json['data']
-    if len(payload) > 1:
-        return 'Multiple data elements in notifications not supported yet', 400
-    payload = payload[0]
 
     log().info('Received payload: {}'.format(payload))
 
-    # Validate notification
-    error = _validate_payload(payload)
-    if error:
-        return error, 400
-
-    # Add TIME_INDEX attribute
-    payload[CrateTranslator.TIME_INDEX_NAME] = \
-        select_time_index_value_as_iso(request.headers, payload)
-
-    # Add GEO-DATE if enabled
-    add_geodata(payload)
-
-    # Always normalize location if there's one
-    normalize_location(payload)
+    # preprocess and validate each entity update
+    for entity in payload:
+        # Validate entity update
+        error = _validate_payload(entity)
+        if error:
+            return error, 400
```

Comment:
This seems not to be possible or at least not simple. Only things _validate_payload reports errors about are missing type or id but Connexion alrady checks them before an gives its own error which only reports the issue it noticed first. I know nothing about Connexion so I don't know if this could be some how done with it.

Reply:
also, we'll have to think about what to do if validation for a record fails. What if we have a batch of 10 and two entities fail validation? Are we going to ditch the whole lot?! Or should we rather accept the 8 that are valid and return an error message indicating partial failure, possibly with an indication of which entity failed validation and why?

Reply:
Will be covered by https://github.com/smartsdk/ngsi-timeseries-api/issues/193